source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include "envoy/config/filter/udp/udp_proxy/v2alpha/udp_proxy.pb.h"
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/network/filter.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/network/utility.h"
+
+#include "absl/container/flat_hash_set.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace UdpProxy {
+
+class UdpProxyFilterConfig {
+public:
+  UdpProxyFilterConfig(Upstream::ClusterManager& cluster_manager, TimeSource& time_source,
+                       const envoy::config::filter::udp::udp_proxy::v2alpha::UdpProxyConfig& config)
+      : cluster_manager_(cluster_manager), time_source_(time_source), config_(config) {}
+
+  Upstream::ThreadLocalCluster* getCluster() const {
+    return cluster_manager_.get(config_.cluster());
+  }
+  TimeSource& timeSource() const { return time_source_; }
+
+private:
+  Upstream::ClusterManager& cluster_manager_;
+  TimeSource& time_source_;
+  const envoy::config::filter::udp::udp_proxy::v2alpha::UdpProxyConfig config_;
+};
+
+using UdpProxyFilterConfigSharedPtr = std::shared_ptr<const UdpProxyFilterConfig>;
+
+class UdpProxyFilter : public Network::UdpListenerReadFilter, Logger::Loggable<Logger::Id::filter> {
+public:
+  UdpProxyFilter(Network::UdpReadFilterCallbacks& callbacks,
+                 const UdpProxyFilterConfigSharedPtr& config)
+      : UdpListenerReadFilter(callbacks), config_(config) {}
+
+  // Network::UdpListenerReadFilter
+  void onData(Network::UdpRecvData& data) override;
+
+private:
+  class ActiveSession : public Network::UdpPacketProcessor {
```

Comment:
nits: comment about what a session stands for, and how it  proxies packets, i.e. read/write path?